server/api/tvdb/index.ts

```diff
@@ -341,6 +347,137 @@ class Tvdb extends ExternalAPI implements TvShowProvider {
       return this.createEmptySeasonResponse(tvId);
     }
 
+    let wantedTranslation = convertTmdbLanguageToTvdbWithFallback(
+      language,
+      Tvdb.DEFAULT_LANGUAGE
+    );
+
+    // check if translation is available for the season
+    const availableTranslation = season.nameTranslations.filter(
+      (translation) =>
+        translation === wantedTranslation ||
+        translation === Tvdb.DEFAULT_LANGUAGE
+    );
+
+    if (!availableTranslation) {
+      // if no translations are available, use the original language (no wanted translation and no english fallback)
+      wantedTranslation = tvdbData.originalLanguage;
+    }
+
+    if (season && wantedTranslation === tvdbData.originalLanguage) {
```

Comment:
You already check above if season is not empty.
```suggestion
    if (wantedTranslation === tvdbData.originalLanguage) {
```